### PR TITLE
LPS-195732 Merge BrowserSniffer interface and impl logic into Util in kernel

### PR DESCRIPTION
**Pull request by julschong**

### CODE REVIEW

**Comment by catarineaguiar** [comment]
@julschong possible regression:

```
     [exec] com.liferay.portal.url.builder.internal.BrowserModuleAbsolutePortalURLBuilderTest > test[0: context=false, proxy=false, cdnHost=false] FAILED
     [exec]     java.lang.NullPointerException
     [exec]         at com.liferay.portal.url.builder.internal.BaseAbsolutePortalURLBuilderTestCase.setUp(BaseAbsolutePortalURLBuilderTestCase.java:160)
     [exec]         at com.liferay.portal.url.builder.internal.BrowserModuleAbsolutePortalURLBuilderTest.setUp(BrowserModuleAbsolutePortalURLBuilderTest.java:54)
     [exec]         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [exec]         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [exec]         at java.lang.reflect.Method.invoke(Method.java:498)
     [exec]         at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-09/test-1-40/test-portal-acceptance-pullrequest(master)/5897/modules-unit-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
no regressions found

**Comment by julschong** [comment]
changed variable name back in TemplateContextHelper

**Comment by julschong** [comment]
ci:test:sf

File: portal-kernel/src/com/liferay/portal/kernel/servlet/BrowserSnifferUtil.java [hunk 1]
```diff
@@ -15,120 +23,443 @@
...
+
+	public static final String BROWSER_ID_EDGE = "edge";
+
+	public static final String BROWSER_ID_FIREFOX = "firefox";
+
+	public static final String BROWSER_ID_IE = "ie";
+
+	public static final String BROWSER_ID_OTHER = "other";
+
+	public static final String[] REVISION_LEADINGS = {
```
Comment (shuyangzhou) [inline portal-kernel/src/com/liferay/portal/kernel/servlet/BrowserSnifferUtil.java:36]: @julschong no reason to expose these constants array as public. They are only ever referenced by tests, keep them private, tests can use reflection to get those fields.

**Comment by shuyangzhou** [comment]
@julschong see comment above, please resend.

**Pull request closed without merging**